### PR TITLE
Restore failing CircleCI tests on error

### PR DESCRIPTION
**Pull request by yvanzo**
# Problem

Before the commit a0b56e9f8f9729f1e0c33df799ddbdaf29d0a44b the commands in the `js-perl-and-pgtap` test were ran with these shell options:

* `-e` to exit on error status code of any command,
* `-o pipefail` to exit on error status code of any piped sub-command.

See https://circleci.com/docs/configuration-reference/#default-shell-options
    
However the new script didn’t have these options enabled, causing failures from `flow` (among others) to go unnoticed.

# Solution

This patch restores the previous/wanted behavior by enabling these in the script.

I built and uploaded a new pair of test images with the version [`v-2024-01.419`](https://hub.docker.com/r/metabrainz/musicbrainz-tests/tags?page=1&name=v-2024-01.419).

I also resolved the failures that went unnoticed so far:
- ESLint errors
- Genie and Bugs! URL web tests
- MBS-9669 Selenium test

See commit messages for details.

# Further action

Rebase on `master` any PR that need to be tested with the new images.

### CODE REVIEW

**Comment by yvanzo** [comment]
The job [18519](https://app.circleci.com/pipelines/github/metabrainz/musicbrainz-server/10712/workflows/4fd242a0-d906-4b1b-a995-4e4b994cd862/jobs/18519) successfully failed as it should have failed for `master` before the merge of the pull request #3144.

**Comment by yvanzo** [comment]
Rebased on `master` including the fix of Flow tests  from the merged pull request #3144.

**Comment by yvanzo** [comment]
Fixed ESLint errors that went unnoticed for the last months.

**Comment by yvanzo** [comment]
@brainzbot please retest this

**Comment by yvanzo** [comment]
Forgot to submit. PR description now updated.

**Review by mwiencek — APPROVED** [review]
Thanks for fixing up this mess!

File: lib/MusicBrainz/Server/Translation.pm [hunk 1]
```diff
@@ -131,7 +131,7 @@ sub set_language
     }
     my $set_lang = web_set_locale(\@avail_lang, [ 'utf-8' ], LC_MESSAGES);
     if (!defined $set_lang) {
-        return 'en';
+        return contains_string(\@avail_lang, 'es_419') ? 'es_419' : 'en';
```
[the remark this answers is not in the document]
Comment (yvanzo) [reply lib/MusicBrainz/Server/Translation.pm:137]: My guess is that `Locale::Util` might not handle `es_419` code. I also noticed that there is no system locale `es_419.utf8`. But I didn’t dig it fully and opened MBS-13446 instead as we don’t really need it to work for now and it seems to be a rabbit hole.

Updated this change with a comment referring to this ticket. See commit https://github.com/metabrainz/musicbrainz-server/pull/3146/commits/2d70c076b28cabaef34529c79f2248eb6ac235e7.

File: lib/MusicBrainz/Server.pm [hunk 2]
```diff
@@ -347,7 +347,7 @@ sub with_translations {
     my $cookie_lang = Translation->instance->language_from_cookie($c->request->cookies->{lang});
     $c->set_language_cookie($c->request->cookies->{lang}->value) if defined $c->request->cookies->{lang};
     my $lang = Translation->instance->set_language($cookie_lang);
-    my $html_lang = $lang =~ s/_([A-Z]{2})/-\L$1/r;
+    my $html_lang = $lang =~ s/_([A-Z0-9]{2,})/-\L$1/r;
```
[the remark this answers is not in the document]
Comment (yvanzo) [reply lib/MusicBrainz/Server.pm:350]: There isn’t any risk with very long variant strings that contain only characters in the range `[A-Z0-9]`.

**Comment by yvanzo** [comment]
> Seems good, although I'd like a bit more of an explanation (well, any, really :) ) for the "Keep tests silent when service is running already" commit.

Rewrote the message for commit https://github.com/metabrainz/musicbrainz-server/pull/3146/commits/c13df4c0beee4e806f4d1b559cd5d213060365e1

**Comment by yvanzo** [comment]
Selenium build [#10264](https://ci.metabrainz.org/job/musicbrainz-server/10264/) finished successfully whereas later builds  [#10266](https://ci.metabrainz.org/job/musicbrainz-server/10266/) and [#10273](https://ci.metabrainz.org/job/musicbrainz-server/10273/) failed for two different obscure unrelated reasons. Those were running the same changes beyond an added comment and a rephrased commit message.

**Pull request merged**